### ISSUE TITLE
Use events

### ISSUE THREAD

**Issue by jtojnar** [Push-EDX/flarum-ext-chat#30]
So that it could be hooked into [notify](https://github.com/manelizzard/flarum-notify).

**Comment by gpascualg** [comment]
Maybe only for mentions with @? I don't clearly see how we would go about it though, as I see some problems:

* You might get an insane amount of notifications: Users should be able to toggle whether they want or not to receive chat notifications
* In case you do want them, the notification itself should contain the message itself? Clicking it should maybe bring you to the specific line on the chat? There's quite a lot of work

Good idea though!

**Comment by jtojnar** [comment]
My use case is a Telegram chat for moderators, where every action on the fora is posted, so that moderators could react quickly if there is some problem.

Since the chat is common for all moderators, it would not make sense to only notify about mentions. Or there would need to be an option for selecting the users/user group, whose mention would trigger the notification.

For convenience, the message text should be included, so that moderators do not need to visit the site each time a new message is posted, even if it is okay. Link to the specific chat line would be nice, but it is not a necessity – for now, the message still needs to be erased in the database.

In conclusion, the only critical thing is triggering an event when a message is posted. There is a nice example in [flagrow/upload](https://github.com/flagrow/upload/blob/332d61edd45d4fb00159aef0ee5e15918912ac6e/src/Commands/UploadHandler.php#L97).

**Comment by gpascualg** [comment]
Oh my bad, I completely misunderstood you. I thought you meant notifications, as in the "new answer in post" notification.

That completely changes it, and it is much more easy and direct to do. I agree it would be nice to have such functionality!
